Form/Type/DateRangeType.php

```diff
@@ -20,15 +20,28 @@
 class DateRangeType extends AbstractType
 {
     /**
-     * @var TranslatorInterface
+     * @var TranslatorInterface|null
+     *
+     * @deprecated translator property is deprecated since version 3.x, to be removed in 4.0
      */
     protected $translator;
 
     /**
-     * @param TranslatorInterface $translator
+     * @param TranslatorInterface|null $translator
+     *
+     * @deprecated translator dependency is deprecated since version 3.x, to be removed in 4.0
      */
-    public function __construct(TranslatorInterface $translator)
+    public function __construct(TranslatorInterface $translator = null)
     {
+        // NEXT_MAJOR: make the $translator parameter required
```

Comment:
Shouldn't we just remove the constructor on next major release? I don't understand why this parameter should be required again